providers/freebsd/os_freebsd_test.go

```diff
@@ -0,0 +1,46 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package freebsd
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/elastic/go-sysinfo/types"
+)
+
+func TestOperatingSystem(t *testing.T) {
+	t.Run("freebsd14", func(t *testing.T) {
+		os, err := getOSInfo("")
+		if err != nil {
+			t.Fatal(err)
+		}
+		assert.Equal(t, types.OSInfo{
+			Type:     "",
+			Family:   "freebsd",
+			Platform: "freebsd",
+			Name:     "FreeBSD",
+			Version:  "14.0-RELEASE",
```

Comment:
This test is dependent upon the environment under which is runs which makes it brittle. Can this be refactored to compare the expected values against something returned by another source (e.g. `freebsd_version`).

Reply:
How about some regex?

```go
t.Run("freebsd", func(t *testing.T) {
    os, err := getOSInfo("")
    if err != nil {
        t.Fatal(err)
    }
    assert.IsType(t, types.OSInfo{}, *os)
    assert.Equal(t, "freebsd", os.Type)
    assert.Equal(t, "freebsd", os.Family)
    assert.Equal(t, "freebsd", os.Platform)
    assert.Equal(t, "FreeBSD", os.Name)
    assert.Regexp(t, regexp.MustCompile("([0-9]{1,2})\\.([0-5])-(RELEASE|STABLE|CURRENT|RC[0-9])(-p[0-9])?"), os.Version)
    assert.Regexp(t, regexp.MustCompile("([0-9]){1,2}"), os.Major)
    assert.Regexp(t, regexp.MustCompile("([0-9])"), os.Minor)
    assert.Regexp(t, regexp.MustCompile("([0-9]){1,2}"), os.Patch)
    t.Logf("%#v", os)
})
```

Reply:
Works for me. It's less brittle than was there.

Reply:
also "ALPHA" and "BETA" should be in that regex. Other than that it's fine.

Reply:
> also "ALPHA" and "BETA" should be in that regex. Other than that it's fine.

I mean `ALPHA([0-9]{0,2})|BETA([0-9]{0,2})` in `os.Version` and `([0-9]{1,2})` in `os.Minor`.

Reply:
> > also "ALPHA" and "BETA" should be in that regex. Other than that it's fine.
> 
> I mean `ALPHA([0-9]{0,2})|BETA([0-9]{0,2})` in `os.Version` and `([0-9]{1,2})` in `os.Minor`.

Gotcha; the code has been updated.

Reply:
Added 85cdffb